src/main/java/uk/gov/hmcts/darts/arm/service/impl/AbstractArmBatchProcessResponseFiles.java

```diff
@@ -0,0 +1,836 @@
+                                                 ArmResponseUploadFileRecord armResponseUploadFileRecord,
+                                                 ExternalObjectDirectoryEntity externalObjectDirectory,
+                                                 String objectChecksum) {
+        externalObjectDirectory.setExternalFileId(armResponseUploadFileRecord.getA360FileId());
+        externalObjectDirectory.setExternalRecordId(armResponseUploadFileRecord.getA360RecordId());
+        externalObjectDirectory.setDataIngestionTs(OffsetDateTime.now());
+        updateExternalObjectDirectoryStatus(externalObjectDirectory, EodHelper.storedStatus());
+    }
+
+    protected void onUfChecksumValidationFailure(ArmResponseUploadFileRecord armResponseUploadFileRecord,
+                                                 ExternalObjectDirectoryEntity externalObjectDirectory,
+                                                 String objectChecksum) {
+        log.warn("External object id {} checksum differs. Arm checksum: {} Object Checksum: {}",
+                 externalObjectDirectory.getId(),
+                 armResponseUploadFileRecord.getMd5(), objectChecksum);
+        externalObjectDirectory.setErrorCode(armResponseUploadFileRecord.getErrorStatus());
+        updateExternalObjectDirectoryStatus(externalObjectDirectory, EodHelper.armResponseChecksumVerificationFailedStatus());
+    }
+
+    protected void processInvalidLineFile(ArmResponseInvalidLineRecord armResponseInvalidLineRecord,
+                                          ExternalObjectDirectoryEntity externalObjectDirectory) {
+        // Read the invalid lines file and log the error code and description with EOD
+        log.warn(
+            "ARM invalid line for external object id {}. ARM error description: {} ARM error status: {}",
+            externalObjectDirectory.getId(),
+            armResponseInvalidLineRecord.getExceptionDescription(),
+            armResponseInvalidLineRecord.getErrorStatus()
+        );
+        updateTransferAttempts(externalObjectDirectory);
+        externalObjectDirectory.setErrorCode(armResponseInvalidLineRecord.getExceptionDescription());
+        updateExternalObjectDirectoryStatus(externalObjectDirectory, EodHelper.armResponseManifestFailedStatus());
+    }
+
+    private UploadNewFileRecord readInputJson(String input) {
+        UploadNewFileRecord uploadNewFileRecord = null;
+        if (StringUtils.isNotEmpty(input)) {
+            String unescapedJson = StringEscapeUtils.unescapeJson(input);
+            try {
+                uploadNewFileRecord = objectMapper.readValue(unescapedJson, UploadNewFileRecord.class);
+            } catch (JsonMappingException e) {
+                log.error("Unable to map the input field {}", e.getMessage());
+            } catch (JsonProcessingException e) {
+                log.error("Unable to parse the upload new file record {}", e.getMessage());
```

Comment:
Same

Reply:
changes to existing code that are not necessary for plugging-in DETS functionality cannot be tackled in this PR, as explained in the PR description